packages/cli/src/utils/platform.ts

```diff
@@ -71,9 +71,9 @@ export const ErrorPlatformUnauthorized = new Error(
 
 export const getPlatformTokenOrThrow = async <$Args extends Record<string, unknown>>(args: $Args) => {
   try {
-    const token =
-      getOptionalParameter(args, ['--token', '-t'], 'PRISMA_TOKEN') ||
-      (await readAuthConfig().catch(() => ({ token: '' }))).token
+    const authJson = await readAuthConfig()
+    if (isError(authJson)) throw authJson
+    const token = getOptionalParameter(args, ['--token', '-t'], 'PRISMA_TOKEN') || authJson.token
```

Comment:
The order gets weird here I think, let me explain:

The order of preference for the token is mentioned in https://github.com/prisma/prisma/pull/22144
1. Attempt to read value from arguments `--token` or `-t`
2. Attempt to read value from environment variable `PRISMA_TOKEN`
3. Attempt to read token stored in `auth.json` on the disk
4. If none of the above is present, throw `ErrorPlatformUnauthorized` 

But here, if `--token` is passed, then the auth config will first be read and could throw.

Reply:
Let me improve this a bit!